File: Androidparty/app/src/main/java/com/tzemaitis/androidparty/PartyApplication.java
package com.tzemaitis.androidparty;

import android.app.Application;

/**
 * Created by tzemaitis on 25/07/18.
 */

public class PartyApplication extends Application {

    private TesonetApiInterface api;

    @Override
    public void onCreate() {
        super.onCreate();

        api = TesonetApiClient.getClient().create(TesonetApiInterface.class);
    }

    TesonetApiInterface getRestApi() {
        return api;
    }
}

File: app/src/main/java/com/edvinas/balkaitis/party/app/PartyApplication.kt
package com.edvinas.balkaitis.party.app

import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import timber.log.Timber

class PartyApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> = DaggerAppComponent.builder().create(this)

    override fun onCreate() {
        super.onCreate()
        Timber.plant(Timber.DebugTree())
    }
}
